Лабораторная работа 5

flamangoo opened the pull request.

flamangoo committed:
Лабораторная работа 5

flamangoo committed:
Update task_5.py

flamangoo committed:
Update task_3.py

flamangoo committed:
Update task_3.py

flamangoo committed:
Update task_3.py

andreiivanov-wa commented on lab5/task_3.py:
```diff
@@ -0,0 +1,16 @@
+
+def get_unique_list_numbers(begin: int = -10, end: int = 10, length: int = 15) -> list[int]:
+    list_ = []
+    while len(list_) < length:  # список заполняется до заданной длины
```
здесь бесконечный цикл получится, если length будет задан больше количества целых чисел между begin и end

andreiivanov-wa commented on lab5/task_5.py:
```diff
@@ -0,0 +1,11 @@
+from random import sample
+
+
+def get_random_password(n: int = 8) -> str:
```
n -- не самое удачное название аргумента функции, лучше length или size

flamangoo committed:
Update task_3.py

По материалам лекции: чтобы проверить условие и предотвратить бесконечный цикл, можно вызвать ошибку, явно указывающую на неправильно заданную длину. В таком случае, думаю, имеет смысл проверить и границы, чтобы ошибка не была одинаковой: либо вызвать еще одну ошибку, указывающую на неправильный диапазон, либо автоматически поменять границы множественным присваиванием

flamangoo committed:
Update task_5.py

Исправлено название переменной

flamangoo commented on lab5/task_3.py:
```diff
@@ -0,0 +1,21 @@
+    if start > end:
+        raise ValueError('Левая граница не может быть больше правой')
+        # start, end = end, start
+    if length > end - start:
+        raise ValueError('Длина списка не может быть больше количества случайных целых чисел')
```
о материалам лекции: чтобы проверить условие и предотвратить бесконечный цикл, можно вызвать ошибку, явно указывающую на неправильно заданную длину. В таком случае, думаю, имеет смысл проверить и границы, чтобы ошибка не была одинаковой: либо вызвать еще одну ошибку, указывающую на неправильный диапазон, либо автоматически поменять границы множественным присваиванием

flamangoo replied:
По материалам лекции: чтобы проверить условие и предотвратить бесконечный цикл, можно поднять ошибку, явно указывающую на неправильно заданную длину. В таком случае, думаю, имеет смысл проверить и границы, чтобы сообщение об ошибке не дублировалось: либо вызвать еще одну ошибку, указывающую на неправильный диапазон, либо автоматически поменять границы множественным присваиванием, если пользователь, например, перепутал левую и правую границу. Наверное, это зависит от задачи

flamangoo commented on lab5/task_5.py:
```diff
@@ -0,0 +1,11 @@
+    return password
+
+
+print(get_random_password())
```
Исправлено название переменной
